llvm/lib/Analysis/CaptureTracking.cpp

```diff
@@ -359,6 +359,11 @@ UseCaptureInfo llvm::DetermineUseCaptureKind(const Use &U, const Value *Base) {
   case Instruction::AddrSpaceCast:
     // The original value is not captured via this if the new value isn't.
     return UseCaptureInfo::passthrough();
+  case Instruction::PtrToAddr:
+    // FIXME: the following does not work as expected, so just assume address
+    // is always captured:
+    //  return UseCaptureInfo(CaptureComponents::None, CaptureComponents::Address);
```

Comment:
With this change, `captures_ptrtoaddr_ignored` gives me `captures(none)` but for some reason `captures_ptrtoaddr_ret ` gives `captures(ret: address, provenance)` and `captures_ptrtoaddr_stored` doesn't infer any attribute anymore.

I assume I should just use the current `CaptureComponents::Address;`?

Reply:
Thanks for confirming this! Somehow skipped over that section of LangRef. Will update